forcedotcom/source-deploy-retrieve
# Sm/registry-children

### What does this PR do?
1. add missing `childTypes` properties
2. UT to validate the `types` with a `children` property have a matching entry in `childTypes`

### What issues does this PR fix or reference?
@W-9951980@
https://github.com/forcedotcom/cli/issues/1199
@W-9951980@
https://github.com/forcedotcom/cli/issues/1200
W-9952021

QA Notes:
Besides testing that this fixes the 2 types named in the github issue,  
make sure that the other childTypes added in registryjson don't cause problems deploy/retrieve/converting their metdata.  They were added because the registryValidation test flagged them as missing.
